src/libstore/local-store.cc

```diff
@@ -697,6 +697,17 @@ void canonicalisePathMetaData(const Path & path,
 }
 
 
+void discardNar(Source& source, const ValidPathInfo& info) {
+    int count = info.narSize / 4096;
+    char buffer[4096];
+    for (int i=0; i < count; ++i) {
+        source(buffer, 4096);
+    }
+    int extra = info.narSize % 4096;
+    source(buffer, extra);
+}
```

Comment:
Can you extract the `discard`part of it in `serialise.{cc,hh}`? (maybe even add this as a method of `Source` since it seems to be a generally useful thing to have) and just have this be a thin wrapper like `source.discard(info.narSize)`?

Reply:
Alternatively we could maybe properly parse the Nar before discarding it. That would be marginally slower but also be more resilient in case of a corrupted stream. @edolstra , any opinion?

Reply:
Sounds very sensible to me.

We could not come up with a "easy self-contained reproducer", it would have been nice indeed. If there's an easy way of creating a dummy store for LocalStore then we can create some data to run addToStore with and mock isValidPath so we end up in the discard code.

Reply:
Yes, it's better to use
```
                    ParseSink sink; /* null sink; just parse the NAR */
                    parseDump(sink, source);
```
This also has the advantage that it doesn't require `info.narSize` to be correct. (IIRC there are some legacy cases where it can be zero.)